src/core/outlet_observer.ts

```diff
@@ -113,8 +113,8 @@ export class OutletObserver implements SelectorObserverDelegate {
 
   // Private
 
-  private selector(outletName: string) {
-    return this.scope.outlets.getSelectorForOutletName(outletName)
+  private outletAttributeFor(outletName: string) {
+    return this.context.schema.outletAttributeForScope(this.identifier, outletName)
```

Comment:
I'm wondering if it makes sense that the `SelectorObserver` takes in an `attributeName` to extract the `selector` from instead of passing in the `selector` directly.

This makes the `SelectorObserver` dependent on an element/attribute to be present in order to be able to extract the selector from, so you couldn't just pass in a selector string anymore.

We currently don't have the need for this in the codebase, but I guess it would make sense to keep the responsibilities separated.

Reply:
Are you suggesting that the `OutletObserver` instance should be the one to listen for changes to the attribute, then somehow communicate with the `SelectorObserver` to refresh its set of elements?

Reply:
I haven't actually thought this far. I was just confused that the `SelectorObserver` doesn't get an actual selector but needs to dig it out from somewhere, which technically shouldn't be it's responsibility, since it _could_ be used otherwise.

But now that you mentioned it, I guess it should be the responsibility of the `SelectorObserver `. Maybe we can extract the common logic so that the `OutletObserver` doesn't need to communicate with the `SelectorObserver ` to refresh the elements.

Reply:
> Maybe we can extract the common logic so that the `OutletObserver` doesn't need to communicate with the `SelectorObserver`  to refresh the elements.

The changes proposed in https://github.com/hotwired/stimulus/pull/627 would benefit from a shared abstraction that watches an attribute with `document`-wide implications.

Reply:
I implemented my proposal as discussed on Discord in https://github.com/hotwired/stimulus/pull/624/commits/4ff1250a04685ca9471f38cb37e3a06e750c1bf9